lib/jxl/dec_cache.cc

```diff
@@ -195,23 +196,57 @@ Status PassesDecoderState::PreparePipeline(ImageBundle* decoded,
       }
     }
 
+    ColorEncoding linear_color_encoding;
+    if (frame_header.color_transform != ColorTransform::kXYB) {
+      linear_color_encoding = ColorEncoding::LinearSRGB(
+          output_encoding_info.orig_color_encoding.IsGray());
+    } else {
+      linear_color_encoding = output_encoding_info.linear_color_encoding;
+    }
+
     auto tone_mapping_stage = GetToneMappingStage(output_encoding_info);
     if (tone_mapping_stage) {
       if (!linear) {
         auto to_linear_stage = GetToLinearStage(output_encoding_info);
         if (!to_linear_stage) {
```

Comment:
I think maybe I would swap this condition around (`if (!cond) { a } else { b }` -> `if (cond) { b } else { a }`).

Reply:
done